src/common/dao/project.go

```diff
@@ -302,5 +302,8 @@ func GetRolesByLDAPGroup(projectID int64, groupDNCondition string) ([]int, error
 		log.Warningf("Error in GetRolesByLDAPGroup, error: %v", err)
 		return nil, err
 	}
+	if len(roles) == 1 && roles[0] == 0 { //Avoid return 0 as role
```

Comment:
Add more comments to elaborate the reason please.

Reply:
OK